### PR TITLE
Fix authentication error: social profile not found in HTML

### PR DESCRIPTION
**Pull request by davidkroell**
Garmin changed their API some time back, so the authentication had to be changed.

I've got a bug report in this issue davidkroell/bodycomposition#10.

Thanks to @kochka, who came up with a solution to the same problem in is his own project, so i just copied it over.
https://github.com/kochka/WeightLogger/commit/4973711aa92326b527066227f561fed32e076483#diff-5e915b6d9fa63aa88cec51ef4c31a83b6e068acb8389ce66563bae1e35d91694

Kind regards,
David

### CODE REVIEW

**Comment by abrander** [comment]
Thanks for the PR - nice catch and solution.

I'll merge right away if you remove the unneeded parameters. Only `service`, `gauthHost`, `generateExtraServiceTicket` and the new `generateTwoExtraServiceTickets` are needed.

Nice idea splitting the URL into multiple lines, it makes it much more readable, maybe we should move `service` too..?

File: Client.go [hunk 1]
```diff
@@ -377,7 +377,14 @@ func (c *Client) authenticated() bool {
...
 	// called from do() upon session renewal.
-	URL := "https://sso.garmin.com/sso/signin?service=https%3A%2F%2Fconnect.garmin.com%2Fmodern%2F&gauthHost=https%3A%2F%2Fconnect.garmin.com%2Fmodern%2F&generateExtraServiceTicket=true"
+	URL := "https://sso.garmin.com/sso/signin?service=https%3A%2F%2Fconnect.garmin.com%2Fmodern%2F" +
+		"&gauthHost=https%3A%2F%2Fconnect.garmin.com%2Fmodern%2F" +
+		"&generateExtraServiceTicket=true" +
+		"&generateTwoExtraServiceTickets=true" +
+		"&useCustomHeader=false" +
+		"&mfaRequired=false" +
+		"&rememberMyBrowserShown=false" +
+		"&rememberMyBrowserChecked=false"
```
Comment (abrander) [inline Client.go:387]: `useCustomHeader`, `mfaRequired`, `rememberMyBrowserShown` and  `rememberMyBrowserChecked` are unneeded.

File: Client.go [hunk 2]
```diff
@@ -377,7 +377,14 @@ func (c *Client) authenticated() bool {
 func (c *Client) Authenticate() error {
 	// We cannot use Client.do() in this function, since this function can be
 	// called from do() upon session renewal.
-	URL := "https://sso.garmin.com/sso/signin?service=https%3A%2F%2Fconnect.garmin.com%2Fmodern%2F&gauthHost=https%3A%2F%2Fconnect.garmin.com%2Fmodern%2F&generateExtraServiceTicket=true"
+	URL := "https://sso.garmin.com/sso/signin?service=https%3A%2F%2Fconnect.garmin.com%2Fmodern%2F" +
```
Comment (abrander) [inline Client.go:380]: Can we move `service` to its own line? That would put readability through the roof! :100:

**Review by abrander — CHANGES_REQUESTED** [review]
I know this is working as expected. Thanks. I have a few requests to aid in future debugging and readability.

File: Client.go (hunk 2 above)
Comment (davidkroell) [reply Client.go]: I moved the service parameter to a new line and also removed the unneccessary parameters.

Also gave it a quick test - works exactly the same.

**Comment by abrander** [comment]
This looks awesome! I'll merge right away. Thanks! :smiley:

**Review by abrander — APPROVED** [review]

**Pull request merged**